app/build.gradle

```diff
@@ -187,6 +184,34 @@ dependencies {
     androidTestCompile 'com.android.support:support-annotations:25.1.0'
 }
 
+// SHA and timestamp caching courtesy of https://github.com/gdg-x/frisbee/blob/develop/app/build.gradle#L193-L218
+def commitSha() {
```

Comment:
You may consider putting these method in another named gradle file and apply them just like `apply from: 'quality.gradle'` above. Just a suggestion.